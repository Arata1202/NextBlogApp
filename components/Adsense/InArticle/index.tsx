'use client';

import { useEffect } from 'react';
import { usePathname } from 'next/navigation';

const PUBLISHER_ID = '1705865999592590';

declare global {
  interface Window {
    adsbygoogle: { [key: string]: unknown }[];
  }
}

type DisplayProps = {
  slot: string;
  format?: string;
  responsive?: string;
  style?: any;
};

const InArticle = ({ slot, format = 'fluid', responsive = 'false', style }: DisplayProps) => {
  let pathname = usePathname();
  pathname = pathname ? pathname : '';

  useEffect(() => {
    try {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    } catch (err) {
      console.error(err);
    }
  }, [pathname]);

  return (
    <div
      style={{ maxWidth: '100%', overflow: 'hidden' }}
      className="FirstAd"
      key={pathname.replace(/\//g, '-') + '-' + slot}
    >
      <p className="text-center">スポンサーリンク</p>
      <ins
        className="adsbygoogle"
        style={{ display: 'flex', justifyContent: 'center', width: '100%', ...style }}
        data-ad-layout="in-article"
        data-ad-client={`ca-pub-${PUBLISHER_ID}`}
        data-ad-slot={slot}
        data-ad-format={format}
        data-full-width-responsive={responsive}
      />
    </div>
  );
};

export default InArticle;
