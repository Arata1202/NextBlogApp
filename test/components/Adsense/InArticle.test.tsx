import { render } from '@testing-library/react';
import InArticle from '@/components/Adsense/InArticle';

describe('InArticle', () => {
  test('スナップショット（InArticle)', () => {
    const { asFragment } = render(<InArticle slot="" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
