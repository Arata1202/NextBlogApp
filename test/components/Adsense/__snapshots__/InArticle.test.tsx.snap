// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InArticle スナップショット（InArticle) 1`] = `
<DocumentFragment>
  <div
    class="FirstAd"
    style="max-width: 100%; overflow: hidden;"
  >
    <p
      class="text-center"
    >
      スポンサーリンク
    </p>
    <ins
      class="adsbygoogle"
      data-ad-client="ca-pub-1705865999592590"
      data-ad-format="fluid"
      data-ad-layout="in-article"
      data-ad-slot=""
      data-full-width-responsive="false"
      style="display: flex; justify-content: center; width: 100%;"
    />
  </div>
</DocumentFragment>
`;
